### PR TITLE
fix!: refactoring

### PR DESCRIPTION
**Pull request by elshan-eth**
- rn `minRequiredProofsPerEpoch()` to `minProofsPerEpoch()`
- CommitmentStatuses changed 
- rm totalCCFailCount property in CC to 'totalFailCount'
- big fixes
- math refactoring

### CODE REVIEW

File: subgraph/schema.graphql [hunk 1]
```diff
@@ -3,7 +3,7 @@
```
Comment (AlcibiadesCleinias) [inline subgraph/schema.graphql:3]: so, no diffs in this file, right?

File: subgraph/src/contracts.ts [hunk 2]
```diff
@@ -9,13 +9,13 @@ import { ERC20 } from "../generated/Market/ERC20";
 import { Core } from "../generated/Core/Core";
```
Comment (AlcibiadesCleinias) [inline subgraph/src/contracts.ts:9]: so, no diffs in this file?

File: src/core/modules/capacity/interfaces/ICapacity.sol [hunk 3]
```diff
@@ -93,32 +94,52 @@ interface ICapacity is ICapacityConst {
...
         uint256 failedEpoch;
-        uint256 withdrawCCEpochAfterFailed;
-        uint256 remainingFailsForLastEpoch;
+        uint256 remainingFailedUnitsInLastEpoch;
+        uint256 filledRemainingFailedUnitsInLastEpoch;
         uint256 exitedUnitCount;
-        uint256 totalWithdrawnReward;
+    }
+
+    struct CommitmentPrograss {
```
Comment (AlcibiadesCleinias) [inline src/core/modules/capacity/interfaces/ICapacity.sol:139]: Progr**e**ss?

File: src/core/modules/capacity/interfaces/ICapacity.sol [hunk 4]
```diff
@@ -93,32 +94,52 @@ interface ICapacity is ICapacityConst {
...
-        uint256 snapshotEpoch;
         uint256 startEpoch;
+    }
+
+    struct CommitmentFinish {
         uint256 failedEpoch;
-        uint256 withdrawCCEpochAfterFailed;
-        uint256 remainingFailsForLastEpoch;
+        uint256 remainingFailedUnitsInLastEpoch;
+        uint256 filledRemainingFailedUnitsInLastEpoch;
```
Comment (AlcibiadesCleinias) [inline src/core/modules/capacity/interfaces/ICapacity.sol:135]: what **filled**  actually means?

File: src/core/modules/capacity/CapacityConst.sol [hunk 5]
```diff
@@ -203,56 +218,54 @@ contract CapacityConst is BaseModule, OwnableUpgradableDiamond, ICapacityConst {
     // #endregion ------------------ External View Functions ------------------
 
     // #region ------------------ External Mutable Functions ------------------
-    function setFLTPrice(uint256 fltPrice_) public onlyOwner {
+    function setFLTPrice(uint256 fltPrice_) public onlyCoreOwner {
```
Comment (AlcibiadesCleinias) [inline src/core/modules/capacity/CapacityConst.sol:221]: do we have FLIP on this?

File: ts-client/src/dealCliClient/indexerClient/queries/offers-query.generated.ts [hunk 6]
```diff
@@ -34,7 +33,7 @@ export const OfferQueryDocument = gql`
 export type SdkFunctionWrapper = <T>(action: (requestHeaders?:Record<string, string>) => Promise<T>, operationName: string, operationType?: string, variables?: any) => Promise<T>;
 
 
-const defaultWrapper: SdkFunctionWrapper = (action, _operationName, _operationType, variables) => action();
+const defaultWrapper: SdkFunctionWrapper = (action, _operationName, _operationType, _variables) => action();
```
Comment (AlcibiadesCleinias) [inline ts-client/src/dealCliClient/indexerClient/queries/offers-query.generated.ts:36]: oh, but it is a generated file, why it is changed? Do nothing, just curious that we have different generators

File: subgraph/src/contracts.ts [hunk 7]
```diff
@@ -82,43 +82,56 @@ export function getMinRequiredProofsPerEpoch(contractAddress: Address): BigInt {
...
-            remainingFails = remainingFails - activeUnitCount;
-            activeUnitCount += nextAdditionalActiveUnitCount;
-
-            failedEpoch = UNO_BIG_INT + lastSnapshotEpoch + (remainingFails / activeUnitCount);
-        }
-
-        // CUrrently it is not used. Artifact from contract side.
-        // const remainingFailsForLastEpoch = remainingFails % activeUnitCount;
-        return failedEpoch;
+  maxFailedRatio: BigInt,
```
Comment (AlcibiadesCleinias) [inline subgraph/src/contracts.ts:85]: as I see the diff is big, but I do not see that this methods works as it should be: mirror logic of _failedEpoch from Capacity: https://github.com/fluencelabs/deal/pull/343/files#diff-13309d9b864df15406e4eec89c67209ddae42038f48cb347b67bccdd3be666d7R83

It seemed that _failedEpoch is deprecated? or I just have been lost ( sorry

**Review by AlcibiadesCleinias — COMMENTED** [review]
After reading diff I want to reread more, but for now I also leave this:

- this method should be changed as well in accordance new statuses: https://github.com/fluencelabs/deal/blob/main/ts-client/src/dealExplorerClient/rpcClients/serializers.ts#L43
- _failedEpoch should be deprecated in `subgraph/` and new logic should have a place

Also, I have not enough time to add tests on offchain matcher to CI/CD. I need to run them with this diff. I hope we could sync our times and run offcain matcher tests as well. Or I will show you how to run (no good docs currently for this, sorry)

File: subgraph/schema.graphql [hunk 8]
```diff
@@ -203,7 +202,7 @@ type Deal @entity {
 
   # Add Statistics below.
   withdrawalSum: BigInt!
-  depositedSum: BigInt!  # calculated from the withdraw() [no rewards withdraw].
+  depositedSum: BigInt! # calculated from the withdraw() [no rewards withdraw].
```
[the remark this answers is not in the document]
Comment (elshan-eth) [reply subgraph/schema.graphql:205]: Seems like we have a different code formatter

File: subgraph/schema.graphql (hunk 1 above)
Comment (elshan-eth) [reply subgraph/schema.graphql:3]: renamed totalCUFailCount to totalFailCount

File: subgraph/src/contracts.ts (hunk 2 above)
Comment (elshan-eth) [reply subgraph/src/contracts.ts:9]: Have with failedEpoch fundtion

File: subgraph/src/contracts.ts (hunk 7 above)
Comment (elshan-eth) [reply subgraph/src/contracts.ts:85]: _failedEpoch works incorrect in contracts side and indexer. I remove _failedEpoch from contracts and fix logic in indexer

File: src/core/modules/capacity/interfaces/ICapacity.sol (hunk 4 above)
Comment (elshan-eth) [reply src/core/modules/capacity/interfaces/ICapacity.sol:135]: restFailedUnitsInLastEpoch are the remaining units that we need to slashed in the last epoch.

fillRemainingFailedUnitsInLastEpoch - this is how much we have already slashed unit from restFailedUnitsInLastEpoch

File: src/core/modules/capacity/CapacityConst.sol (hunk 5 above)
Comment (elshan-eth) [reply src/core/modules/capacity/CapacityConst.sol:221]: FLIP-26

File: ts-client/src/dealCliClient/indexerClient/queries/offers-query.generated.ts (hunk 6 above)
Comment (elshan-eth) [reply ts-client/src/dealCliClient/indexerClient/queries/offers-query.generated.ts:36]: I think we need to sync our code formatter

**Pull request merged**